streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsProducer.java

```diff
@@ -52,39 +51,20 @@
 
     private final Producer<byte[], byte[]> producer;
     private final String applicationId;
-    private final TaskId taskId;
-    private final String logMessage;
     private final boolean eosEnabled;
 
     private boolean transactionInFlight = false;
     private boolean transactionInitialized = false;
 
-    public StreamsProducer(final LogContext logContext,
-                           final Producer<byte[], byte[]> producer) {
-        this(logContext, producer, null, null);
-    }
-
-    public StreamsProducer(final LogContext logContext,
-                           final Producer<byte[], byte[]> producer,
-                           final String applicationId,
-                           final TaskId taskId) {
-        if ((applicationId != null && taskId == null) ||
-            (applicationId == null && taskId != null)) {
-            throw new IllegalArgumentException("applicationId and taskId must either be both null or both be not null");
-        }
-
+    public StreamsProducer(final Producer<byte[], byte[]> producer,
+                           final boolean eosEnabled,
+                           final LogContext logContext,
+                           final String applicationId) {
```

Comment:
This is a bit on the side, but there was some wacky stuff going on in here, and afaict, the only purpose of all the nullable fields was to allow including the task id in exception messages. Do we really need to do that? If so, I'll just add the logContext to the exception message instead, since it already has the task id in it.

Reply:
Actually the `logContext.logPrefix()` should have the format of `stream-thread [%s] task [%s]` already, so all the log4j entries are good. For exception messages, we can just get the `prefix` and then encode that into the exception message.

Reply:
Yep, that's what I was thinking. I'll go ahead and do it.